m2e-maven-runtime/pom.xml

```diff
@@ -47,146 +50,212 @@
-					<artifactId>maven-compiler-plugin</artifactId>
-					<version>3.8.1</version>
-					<configuration>
-						<!-- http://maven.apache.org/plugins/maven-compiler-plugin/ -->
-						<source>11</source>
-						<target>11</target>
-					</configuration>
-				</plugin>
 				<plugin>
 					<groupId>org.apache.felix</groupId>
 					<artifactId>maven-bundle-plugin</artifactId>
 					<version>5.1.1</version>
-					<extensions>true</extensions>
 					<configuration>
+						<supportedProjectTypes>eclipse-plugin</supportedProjectTypes>
+						<!-- PDE does not honor custom manifest location -->
+						<manifestLocation>META-INF</manifestLocation>
+						<niceManifest>true</niceManifest>
 						<instructions>
 							<Embed-Dependency>*;scope=compile|runtime</Embed-Dependency>
 							<Embed-Transitive>true</Embed-Transitive>
-							<Embed-Directory>jars</Embed-Directory>
+							<Embed-Directory>${dependency.jars.folder}</Embed-Directory>
 
 							<_failok>true</_failok>
 							<_nouses>true</_nouses>
 							<_nodefaultversion>true</_nodefaultversion>
-							<_snapshot>${commitDate}</_snapshot>
+							<_snapshot>qualifier</_snapshot>
 
 							<Bundle-SymbolicName>${project.artifactId};singleton:=false</Bundle-SymbolicName>
 							<Bundle-RequiredExecutionEnvironment>JavaSE-11</Bundle-RequiredExecutionEnvironment>
 							<Bundle-Name>%Bundle-Name</Bundle-Name>
 							<Bundle-Vendor>%Bundle-Vendor</Bundle-Vendor>
 							<Bundle-ClassPath>.,{maven-dependencies}</Bundle-ClassPath>
+							<Automatic-Module-Name>${project.name}</Automatic-Module-Name>
 
 							<Import-Package>!*</Import-Package>
 							<Eclipse-BundleShape>dir</Eclipse-BundleShape>
 						</instructions>
-						<archive>
-							<addMavenDescriptor>false</addMavenDescriptor>
-						</archive>
 					</configuration>
 				</plugin>
+				<plugin>
+					<groupId>org.apache.maven.plugins</groupId>
+					<artifactId>maven-antrun-plugin</artifactId>
+					<version>3.0.0</version>
+				</plugin>
 			</plugins>
 		</pluginManagement>
 	</build>
 
 	<profiles>
 		<profile>
-			<id>m2e</id>
-			<activation>
-				<property>
-					<name>m2e.version</name>
-				</property>
-			</activation>
-			<properties>
-				<osgi-version-qualifier>qualifier</osgi-version-qualifier>
-			</properties>
+			<id>generate-osgi-metadata</id>
 			<build>
 				<pluginManagement>
 					<plugins>
+						<!-- Disable Tycho plug-ins that would fail in the first pure Maven build -->
 						<plugin>
-							<groupId>org.apache.felix</groupId>
-							<artifactId>maven-bundle-plugin</artifactId>
-							<configuration>
-								<!-- PDE does not honour custom manifest location -->
-								<manifestLocation>META-INF</manifestLocation>
-							</configuration>
+							<groupId>org.eclipse.tycho</groupId>
+							<artifactId>tycho-packaging-plugin</artifactId>
+							<version>${tycho-version}</version>
+							<executions>
+								<execution>
+									<id>default-build-qualifier</id>
```

Comment:
How is qualifier set then?

Reply:
The qualifier is set in the second build, when the runtime modules are build as ordinary Eclipse plug-ins that embed jars.
The profile `generate-osgi-metadata` is only activated for the first build.
The first Maven build, that is also executed in the IDE, is a 'Maven-mode' build and only downloads all dependency jars into the lib folder of each runtime plugin, generates the `MANIFEST.MF` and the `.classpath` files. So a build-qualifier is not necessary here.

I have tried to describe the relationship between the two build steps in `m2e-maven-runtime/README.md`.

Reply:
OK, thanks. Would you mind making adding a comment in the pom  to explains it?

Reply:
I added a reference to the co-located README.md in the beginning of the profile.